### PR TITLE
ELECT-1580: Clarity parse

### PR DESCRIPTION
**Pull request by anthonyjpesce**
## Description

This PR is more of a draft to get edits/discussion from @st421, but aims to write a quick converter for Clarity election results. There are a few issues I want to chat about:

- Right now it will only work for precincts. Getting county level results working should be a pretty minimal change, but I want to talk about implementation.
- I haven't yet added a method to aggregate multiple county XML files into a consolidated feed. Right now, it will just return a list of parsed feeds if multiple XML files are passed to the converter. Want to talk about final implementation to see if we need it before I go there.
- Will add unit tests once we make sure this conceptually works.

## Jira Ticket

https://arcpublishing.atlassian.net/browse/ELECT-1580

## Test Steps

> elexclarity 105369 GA --level=precinct --filename="tests/fixtures/atkinson_precincts_11-3.xml"
> elexclarity 105369 GA --level=precinct"

### CODE REVIEW

File: src/elexclarity/convert.py [hunk 1]
```diff
@@ -1,7 +1,132 @@
...
+            "counts": self.get_total_votes_from_choices(choices)
+        }
+
+    def transform_result_object(self, result):
+        county = result["Region"]
+        fips = self.county_lookup.get(county)
+        contests = [self.transform_contest(i, fips=fips) for i in result["Contest"]]
+        return {i["name"]: i for i in contests}
+
+    def convert(self, data, statepostal="", outputType="results", style="default", resultsBy=None, **kwargs):
```
Comment (st421) [inline src/elexclarity/convert.py:113]: I know I am the queen of forcing everything into OOP, but the [existing setup](https://github.com/WPMedia/elex-data-importer/blob/develop/elexdata/importer.py#L3) is expecting not an instance method

File: src/elexclarity/cli.py [hunk 2]
```diff
@@ -55,17 +60,9 @@ def cli(electionid, statepostal, filename=None, style="default", outputType="res
...
-        style=style,
-        outputType=outputType,
-        resultsBy=resultsBy,
-        **kwargs)
-    '''
-
-    if outputType in ["summary", "settings"]:
-        result = json.dumps(result, indent=2)
-    print(result)
+    # if outputType in ["summary", "settings"]:
```
Comment (st421) [inline src/elexclarity/cli.py:65]: We should still check for `style=raw` and _not_ do a json dump if we're working with XML

File: src/elexclarity/convert.py (hunk 1 above)
Comment (st421) [reply src/elexclarity/convert.py:113]: I would keep this individual `convert` method here (but detach it from the class), name this class `ClarityXMLPrecinctConverter`, and move it elsewhere (i.e. use [the elex-edison/elex-ap structure](https://github.com/WPMedia/elex-edison/blob/develop/src/elexedison/convert.py#L78) )

**Comment by anthonyjpesce** [comment]
Hmm @st421 one thing I noticed today just going through the data... Gwinnett County in GA keys the races in English and Spanish (on the website and in the XML) https://results.enr.clarityelections.com/GA/Gwinnett/105437/web.264614/#/summary?v=270282%2F

This means 1) Keys/names for races aren't necessarily consistent from county to county and 2) we're going to have to do a little bit of cleanup somewhere in the pipeline.

**Comment by anthonyjpesce** [comment]
> Hmm @st421 one thing I noticed today just going through the data... Gwinnett County in GA keys the races in English and Spanish (on the website and in the XML) https://results.enr.clarityelections.com/GA/Gwinnett/105437/web.264614/#/summary?v=270282%2F
> 
> This means 1) Keys/names for races aren't necessarily consistent from county to county and 2) we're going to have to do a little bit of cleanup somewhere in the pipeline.

Just as a followup to this, the senate keys include:
```
'US Senate (Loeffler) - Special'
'US Senate (Loeffler) - Special Election'
'US Senate (Loeffler) - Special/Senado de los EE.UU. (Loeffler) - Especial'
'US Senate (Perdue)'
'US Senate (Perdue)/Senado de los EE.UU. (Perdue)'
```

**Review by st421 — APPROVED** [review]

**Pull request merged**